sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -373,24 +363,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMDeviceAlloc(
     ContextLock.lock();
   }
 
+  umf_memory_pool_handle_t hPoolInternal = nullptr;
   if (!UseUSMAllocator ||
       // L0 spec says that allocation fails if Alignment != 2^n, in order to
       // keep the same behavior for the allocator, just call L0 API directly and
       // return the error code.
       ((Alignment & (Alignment - 1)) != 0)) {
-    ur_result_t Res =
-        USMDeviceAllocImpl(RetMem, Context, Device, nullptr, Size, Alignment);
-    if (IndirectAccessTrackingEnabled) {
-      // Keep track of all memory allocations in the context
-      Context->MemAllocs.emplace(std::piecewise_construct,
```

Comment:
I think we need to keep the MemAllocs. Those are used to automatically manage lifetime of the allocations I think.

Reply:
Sorry, I thought you removed the code that adds allocations to MemAllocs entirely, but it looks like this was only for case where pool was skipped - your change is OK I think.